packages/aws-cdk/lib/api/aws-auth/sdk-provider.ts

```diff
@@ -90,6 +95,12 @@ export class SdkProvider {
   public static async withAwsCliCompatibleDefaults(options: SdkProviderOptions = {}) {
     const sdkOptions = parseHttpOptions(options.httpOptions ?? {});
 
+    if (options.offline) {
+      sdkOptions.endpoint = 'http://localhost:4566';
```

Comment:
Do you think it would be possibile to allow to override the localstack endpoint with a parameter? There are a couple of reason for that like if someone is using the Pro version (that is cloud hosted) or if you are running localstack in a container

Reply:
Yes of course is possible, I didn't add it as doesn't cover my use-case and also there is another [PR](https://github.com/aws/aws-cdk/pull/7337) that already has the option of passing an endpoint in the CLI

Reply:
Yes, I was asking just because the other PR looks stuck